plugins/chat/db/post_migrate/20221117142910_delete_orphaned_channels.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class DeleteOrphanedChannels < ActiveRecord::Migration[7.0]
+  def up
+    CategoryChannel
+      .joins(
+        "LEFT OUTER JOIN categories ON categories.id = chat_channels.chatable_id AND chat_channels.chatable_type = 'Category'",
+      )
+      .where("categories.id IS NULL")
+      .destroy_all
```

Comment:
Do have a look at our internal development lore about using application code in migration which is not encouraged.

Reply:
Yes definitely need to change this to be SQL only.

Reply:
Indeed, I thought it was targeting migrations and not post migrations where we usually manipulate data but I was wrong I guess :sweat_smile: